File: smart_cafe_kasir/lib/models/meja.dart
class Meja {
  final int? id;
  final int nomorMeja;
  final int kapasitas;
  final String status;
  final String? esp8266Id;

  Meja({
    this.id,
    required this.nomorMeja,
    this.kapasitas = 4,
    this.status = 'kosong',
    this.esp8266Id,
  });

  factory Meja.fromJson(Map<String, dynamic> json) {
    return Meja(
      id: json['id'],
      nomorMeja: json['nomor_meja'],
      kapasitas: json['kapasitas'] ?? 4,
      status: json['status'] ?? 'kosong',
      esp8266Id: json['esp8266_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nomor_meja': nomorMeja,
      'kapasitas': kapasitas,
      'status': status,
      'esp8266_id': esp8266Id,
    };
  }
}

File: backend-laravel/app/Models/Meja.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Meja extends Model
{
    protected $fillable = [
        'nomor_meja', 'kapasitas', 'status', 'esp8266_id'
    ];

    protected $casts = [
    'created_at' => 'datetime:c',
    'updated_at' => 'datetime:c',
    ];

    public function pesananAktif()
    {
    return $this->hasOne(Pesanan::class, 'meja_id')
        ->whereIn('status', ['preparing', 'ready', 'served'])
        ->latest();
    }

}